spec/domain/repository/persist_repository_from_upstream_spec.rb

```diff
@@ -47,41 +47,74 @@
 
       expect(Repository.find(existing_repository.id).attributes.symbolize_keys).to include(**expected_attributes)
     end
+
+    context "with name changed in upstream data" do
+      let(:full_name) { "test/name-change" }
+      let(:raw_upstream_data) { build(:raw_upstream_data, full_name: full_name) }
+      let(:existing_repository) { create(:repository, full_name: full_name) }
+      let(:incoming_repository) { build(:repository) }
+
+      context "upstream update fails" do
+        before do
+          allow(RepositoryHost::Github).to receive(:fetch_repo).with(existing_repository.id_or_name, nil).and_return(nil)
+        end
+
+        it "deletes clashed name" do
```

Comment:
Should there also be an expectation that not only is the thing deleted, but the new thing was created in its place? Seems like those two operations should be atomic/put into a transaction, so that we don't end up in a partial state of "well we deleted the clash but didn't replace it with anything," though if it's preexisting behavior...